packages/uni_app/lib/sigarra/endpoints/api/authentication/login/login.dart

```diff
@@ -7,14 +7,18 @@ import 'package:uni/sigarra/endpoints/api/authentication/login/response.dart';
 import 'package:uni/sigarra/options.dart';
 
 class Login {
-  const Login();
-
-  Future<LoginResponse> call({
-    required String username,
-    required String password,
-    required FacultyRequestOptions? options,
-  }) async {
-    options = options ?? FacultyRequestOptions();
+  const Login({
+    required this.username,
+    required this.password,
+    this.options,
+  });
+
+  final String username;
+  final String password;
+  final FacultyRequestOptions? options;
+
+  Future<LoginResponse> call() async {
```

Comment:
Good work here. Just this small catch, shouldn't this class extend the `Endpoint` as well and override the `call()` method?